packages/forms/src/model.ts

```diff
@@ -45,6 +45,15 @@ export const PENDING = 'PENDING';
  */
 export const DISABLED = 'DISABLED';
 
+/**
+ * A form can have several different statuses:
+ * @see `VALID`
+ * @see `INVALID`
+ * @see `PENDING`
+ * @see `DISABLED`
+ */
+export type FormStatus = 'VALID'|'INVALID'|'PENDING'|'DISABLED';
```

Comment:
I guess you should use the constants to construct the type union
```suggestion
export type FormStatus = VALID|INVALID|PENDING|DISABLED;
```

Reply:
I tried this, but TypeScript doesn't appear to support it:

```
'VALID' refers to a value, but is being used as a type here. Did you mean 'typeof VALID'?
```

Reply:
The `typeof VALID` is indeed needed here; A type alias is parsed in a type-position, whereas `VALID` is only available in the value scope (there is no corresponding declared type `VALID`). Since the value of `VALID` is a string literal that is assigned into a `const` variable, TypeScript will infer its type as the string literal type `'VALID'`. If you want to refer to that implicit type of the value in a type-only position you need the `typeof` operator (which is called a _type query_ in TypeScript terminology) as it accepts a qualified name that is resolved in the value scope.